### PR TITLE
Add Neg strategy and change 4 method names for consistency

### PR DESCRIPTION
**Pull request by amkratz**
Related to [#379](https://github.com/Axelrod-Python/Axelrod/issues/379)

### CODE REVIEW

File: axelrod/strategies/neg.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def strategy(self, opponent):
+        # Random first move
+        if(len(self.history) == 0):
+            return random_choice();
```
Comment (meatballs) [inline axelrod/strategies/neg.py]: This would be better as:
```
if not self.history:
```

File: axelrod/strategies/_strategies.py [hunk 2]
```diff
@@ -46,6 +46,7 @@
 from .mindcontrol import MindController, MindWarper, MindBender
 from .mindreader import MindReader, ProtectedMindReader, MirrorMindReader
 from .mutual import Desperate, Hopeless, Willing
+from .neg import Neg
 from .oncebitten import OnceBitten, FoolMeOnce, ForgetfulFoolMeOnce, FoolMeForever
```
Comment (meatballs) [inline axelrod/strategies/_strategies.py]: The strategy also needs to be added to the all_strategies list in this file

**Comment by meatballs** [comment]
Thanks for the contribution @amkratz ! Just a couple of minor change requests from me.

File: axelrod/strategies/neg.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+C, D = Actions.C, Actions.D
+
+class Neg(Player):
+    """
+    A player starts by cooperating or defecting randomly if it's their first move,
+     then simply doing the opposite of the opponents last move thereafter.
+
+    Names:
+
+    Neg - [No official reference found in docs, only found in 'Desired New Strategies' list: https://github.com/Axelrod-Python/Axelrod/issues/379]
```
Comment (marcharper) [inline axelrod/strategies/neg.py]: Please list this as the reference: http://www.prisoners-dilemma.com/competition.html

Note that the full strategy name is `Negation`, and we should use that as the class name.

File: axelrod/strategies/neg.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    """
+
+    name = "Neg"
```
Comment (marcharper) [inline axelrod/strategies/neg.py]: Negation

File: axelrod/strategies/neg.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    classifier = {
+        'memory_depth': 1,  # Four-Vector = (1.,0.,1.,0.)
```
Comment (marcharper) [inline axelrod/strategies/neg.py]: The memory depth is correct but not the Four-Vector comment, so let's leave that off.

File: axelrod/strategies/neg.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        'stochastic': False,
```
Comment (marcharper) [inline axelrod/strategies/neg.py]: It is stochastic (first move is random).

**Comment by amkratz** [comment]
Thank you for the help, I went ahead and made those changes listed. Sorry if this pull request is kind of messy, it's my first time submitting one.

**Comment by marcharper** [comment]
> Sorry if this pull request is kind of messy, it's my first time submitting one.

Not at all! It's quite good for a first PR on the project and we usually have at least a few minor comments with names and such. Thanks for the contribution!

**Comment by marcharper** [comment]
Looks like a few doc tests are failing now that we've marked the strategy as stochastic. You just need to manually edit the appropriate numbers in the documentation:
```
File "./docs/tutorials/advanced/classification_of_strategies.rst", line 49, in classification_of_strategies.rst

Failed example:
    len(strategies)
Expected:
    37
Got:
    38

----------------------------------------------------------------------

File "./docs/tutorials/advanced/classification_of_strategies.rst", line 60, in classification_of_strategies.rst

Failed example:
    len(strategies)
Expected:
    24
Got:
    25

----------------------------------------------------------------------

File "./docs/tutorials/advanced/classification_of_strategies.rst", line 72, in classification_of_strategies.rst

Failed example:
    len(strategies)
Expected:
    41
Got:
    42
```

This will get the tests working on Travis. @meatballs @drvinceknight :+1: from me once the tests pass!

**Comment by amkratz** [comment]
Everything seems to be in order now, I appreciate the comments. I hope to tackle a few more new strategies soon!

**Comment by meatballs** [comment]
Thanks @amkratz !!

**Pull request merged**